ourjapanlife/findadoc-server
# feat: add mutations for addFacility and addHealthcareProfessional

Resolves #207 
Resolves #211 


# What changed
1. Creates mutation for addFacility that will save a new Facility to Firestore with a list of HealthcareProfessional IDs.
2. Creates mutation for addHealthcareProfessional that will save a new HealthcareProfessional to Firestore.

# Testing instructions
1. Paste the following into the `Variables` section at the bottom of GraphQL studio to create a new Facility and HealthcareProfessional. 
2. Check Firestore and verify that the new documents were created.

```
{
  "input": {
    "nameJa": "some facility ja name",
    "nameEn": "some facility en name",
    "contact": {
      "website": "http://some-website",
      "phone": "000-0000-0000",
      "mapsLink": "http://some-map",
      "email": "some@email.address",
      "address": {
        "addressLine1En": "123 my address",
        "addressLine1Ja": "123アドレス",
        "addressLine2En": "Amazing House",
        "addressLine2Ja": "アメージングハウス",
        "cityEn": "some city en",
        "cityJa": "some city ja",
        "postalCode": "12345",
        "prefectureEn": "some prefecture en",
        "prefectureJa": "some prefecture ja"
      }
    },
    "healthcareProfessionals": [
      {
        "acceptedInsurance": ["INSURANCE_NOT_ACCEPTED", "INTERNATIONAL_HEALTH_INSURANCE"],
        "degrees": [
          {
            "nameJa": "some degree ja",
            "nameEn": "some degree en",
            "id": "1",
            "abbreviation": "S.D."
          }
        ],
        "names": [
          {
            "middleName": "some middle name",
            "locale": "en",
            "lastName": "some last name",
            "firstName": "some first name"
          }
        ],
        "spokenLanguages": [
          {
            "iso639_3": "en",
            "nameEn": "English",
            "nameJa": "English in ja",
            "nameNative": "English"
          }
        ],
        "specialties": [
          {
            "names": [
              {
                "name": "best doctor in the world",
                "locale": "some locale"
              }
            ]
          }
        ]
      }
    ]
  }
}
```
